flutter/alds/lib/mappage.dart

```diff
@@ -214,40 +195,38 @@ class _AldsMapPageState extends State<AldsMapPage> {
   }
 
   List<Widget> _createSavedLocationMarkers() {
-    
-    // Mocked saved locations - (name, (lat, long))
-    List<SavedLocation> savedLocations = [
-      SavedLocation("Home", 41.826874886601985, -71.40318586689112),  // Brown Campus Center
-      SavedLocation("Gym", 41.830156496801976, -71.39804070374443), // Nelson Fitness Center
-      SavedLocation("Work", 41.826922607676, -71.3995623245632), // CIT
-      SavedLocation("Office", 41.82415891316371, -71.39895318840045), // New Watson
-    ];
-
-    List<LocationMarkerLayer> savedLocationMarkers = savedLocations.map((SavedLocation loc) => 
+    return savedLocations.map((SavedLocation loc) => 
       LocationMarkerLayer(
-        position: LocationMarkerPosition(latitude: loc.latitude, longitude: loc.longitude, accuracy: 0.5),
+        position: LocationMarkerPosition(
+          latitude: loc.latitude,
+          longitude: loc.longitude,
+          accuracy: 0.5
+        ),
         style: LocationMarkerStyle(
           marker: const Icon(Icons.location_on, color: Colors.red),
           markerSize: const Size(20, 20),
         ),
       )
     ).toList();
-
-    return savedLocationMarkers;
   }
 
   void _handleValidateLocation() async {
     String txt = _controller.text;
 
-    // Handle invalid input
-    if (txt.isEmpty) {
-      util.log("NO LOCATION ENTERED");
+    if (txt.isEmpty || _curPosition == null) {
+      util.log("NO LOCATION ENTERED OR CURRENT POSITION NOT AVAILABLE");
       return;
     }
 
-    // Validate location
+    await storage.addNewLocation(txt, _curPosition!.latitude, _curPosition!.longitude);
```

Comment:
We don't want to always add a new saved location. If a saved location of the same name already exists, we just want to update that location through validating the location (i.e. running `Locator.merge()`). This is how we guarantee that names of saved locations are unique. Can you make that if you validate an existing location (e.g. selecting one that already exists in the dropdown), the saved location isn't duplicated, rather updated via `merge()`?